### PR TITLE
Feature Request: Search by abbreviated qualified name #1090

### PR DESCRIPTION
**Pull request by TheOneAmir**
Added the ability to have partial search matches when fullNameSearch is false and allowNonContiguousMatch is true. It is possible to use the in-word matching even with the full name search but since the full names are quite long, a lot of irrelevant matches are picked up.

### CODE REVIEW

**Comment by siegfriedpammer** [comment]
Thank you for your contribution!

I noticed that you never set the `allowNonContiguousMatch` field anywhere. How did you plan to activate the feature?

Please note that we are currently moving a lot of things around in ILSpy and that I am unable to merge this in the near future. Once we are done, I will merge and adapt your contribution to the other changes.

Sorry for the wait!

**Comment by TheOneAmir** [comment]
Oh, that's fine. Yeah, I had currently left it as disabled, as I didn't see a good place to add it. You mentioned "I think it would be best to add an option for this" and so I was wondering if you had an idea for where it should be or if I could just add the option where I see fit?

Perhaps by making the search bar smaller and adding an option/dropdown there for searchsettings?
image

**Comment by siegfriedpammer** [comment]
Originally I was thinking about adding a checkbox to the DisplaySettingsPanel to enable or disable the feature in general. Not sure if it's useful to have a short input sequence (like there already is `t:` for types or `/` for regex search) or control to temporarily activate the feature in the search box. To be honest, I haven't thought about that yet. Feel free to come up with a suggestion!

**Comment by TheOneAmir** [comment]
I've temporarily set the activator to be the prefix "s:" so at least the search will be usable/testable. I'm not yet sure what the best way to integrate DisplaySettingsPanel to SearchStrategies would be without introducing unwanted dependencies. If we can confirm that a checkbox is indeed what we want, I can look into this integration.

File: ILSpy/SearchStrategies.cs [hunk 1]
```diff
@@ -114,14 +116,44 @@ protected virtual bool IsMatch(Func<string, string> getText)
 						}
 						break;
 					default:
-						if (text.IndexOf(term, StringComparison.OrdinalIgnoreCase) < 0)
-							return false;
+						if (term.Length > 2 && term.StartsWith(NoncontiguousSearchPrefix, StringComparison.OrdinalIgnoreCase)) {
```
[the remark this answers is not in the document]
Comment (siegfriedpammer) [reply ILSpy/SearchStrategies.cs]: If that's OK with you... ?

**Comment by siegfriedpammer** [comment]
@TheOneAmir Any update on this?

File: ILSpy/SearchStrategies.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheOneAmir) [reply ILSpy/SearchStrategies.cs]: Yes, that makes sense to me. I have just been a little bit occupied with work and I haven't been able to get back around to this. If you're fine waiting the week, I'll have time this weekend.

**Comment by TheOneAmir** [comment]
Alright, I have changed '~' to be the prefix per search term to enable the 'in-string search'. 

It is to be noted that there were a few types and constants which had '~' in them which will no longer be returned. If we wish to retain these searches, the only character for which I didn't find any existing matches is the '@' character (Even + and - have existing matches which are hidden by the prefixed searches).

**Comment by siegfriedpammer** [comment]
Thank you again for your contribution!

**Pull request merged**